pkg/logs/tailers/windowsevent/tailer.go

```diff
@@ -89,210 +94,257 @@ func (t *Tailer) Identifier() string {
+			err = fmt.Errorf("error creating bookmark: %v", err)
+			log.Errorf("%v", err)
+			t.source.Status.Error(err)
+			return
 		}
 	}
 
-	// Normalize the Event.System.EventID field
-	err = normalizeEventID(mv)
+	// subscription
+	t.sub = evtsubscribe.NewPullSubscription(
+		t.config.ChannelPath,
+		t.config.Query,
+		opts...,
+	)
+	err = t.sub.Start()
 	if err != nil {
-		log.Debugf("Error normalizing EventID: %s", err)
-	}
-
-	// Replace Task and Opcode codes by the rendered value
-	if re.task != "" {
-		_, _ = mv.UpdateValuesForPath("Task:"+re.task, taskPath)
-	}
-	if re.opcode != "" {
-		_, _ = mv.UpdateValuesForPath("Opcode:"+re.opcode, opcode)
-	}
-	// Set message and severity
-	if re.message != "" {
-		_ = mv.SetValueForPath(re.message, "message")
-	}
-	if re.level != "" {
-		_ = mv.SetValueForPath(re.level, "level")
+		err = fmt.Errorf("failed to start subscription: %v", err)
+		log.Errorf("%v", err)
+		t.source.Status.Error(err)
+		return
 	}
 
-	jsonEvent, err := mv.Json(false)
+	// render context for system values
+	t.systemRenderContext, err = t.evtapi.EvtCreateRenderContext(nil, evtapi.EvtRenderContextSystem)
 	if err != nil {
-		return &message.Message{}, err
+		err = fmt.Errorf("failed to create system render context: %v", err)
+		log.Errorf("%v", err)
+		t.source.Status.Error(err)
+		return
 	}
-	jsonEvent = replaceTextKeyToValue(jsonEvent)
-	log.Debug("Sending JSON:", string(jsonEvent))
-	return message.NewMessageWithSource(jsonEvent, message.StatusInfo, t.source, time.Now().UnixNano()), nil
-}
+	defer evtapi.EvtCloseRenderContext(t.evtapi, t.systemRenderContext)
 
-// EventID sometimes comes in like <EventID>7036</EventID>
-//
-//	which mxj will transform to "EventID":"7036"
-//
-// other times it comes in like <EventID Qualifiers='16384'>7036</EventID>
-//
-//	which mxj will transform to "EventID":{"value":"7036","Qualifiers":"16384"}
-//
-// We want to normalize this so the resulting JSON is consistent
-//
-//	"EventID":"7036","EventIDQualifier":"16384"
-//
-// Format definition: https://learn.microsoft.com/en-us/windows/win32/wes/eventschema-systempropertiestype-complextype
-func normalizeEventID(mv mxj.Map) error {
-	values, err := mv.ValuesForPath(eventIDPath)
-	if err != nil || len(values) == 0 {
-		return fmt.Errorf("could not find path: %s", eventIDPath)
-	}
-	for _, value := range values {
-		valueMap, ok := value.(map[string]interface{})
-		if !ok {
-			continue
-		}
+	t.source.Status.Success()
 
-		// Get element value
-		text, foundText := valueMap["#text"]
-		// Qualifier optional
-		qualifier, foundQualifier := valueMap["Qualifiers"]
-		if foundText && foundQualifier {
-			// Remove Qualifiers attribute from EventID by
-			// overwriting the path with just the text value
-			_ = mv.SetValueForPath(text, eventIDPath)
-			// Add qualifier value to a new path
-			_ = mv.SetValueForPath(qualifier, eventIDQualifierPath)
-		}
-	}
-	return nil
+	// wait for stop signal
+	t.eventLoop()
+	close(t.done)
```

Comment:
If any of the errors above occur cuasing an early `return`, `t.done` will never be closed and will cause `Stop` block waiting on the channel. Also - as a result the decoder and subscription will never be stopped. I think most of this can be resolved by putting `close(t.done)` in a defer.